core/client/fs/src/main/java/alluxio/util/GrpcDefaultOptions.java

```diff
@@ -0,0 +1,248 @@
+package alluxio.util;
+
+import alluxio.client.ReadType;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
+import alluxio.grpc.CheckConsistencyPOptions;
+import alluxio.grpc.CreateDirectoryPOptions;
+import alluxio.grpc.CreateFilePOptions;
+import alluxio.grpc.DeletePOptions;
+import alluxio.grpc.ExistsPOptions;
+import alluxio.grpc.FileSystemMasterCommonPOptions;
+import alluxio.grpc.FreePOptions;
+import alluxio.grpc.GetStatusPOptions;
+import alluxio.grpc.ListStatusPOptions;
+import alluxio.grpc.LoadMetadataPType;
+import alluxio.grpc.MountPOptions;
+import alluxio.grpc.OpenFilePOptions;
+import alluxio.grpc.RenamePOptions;
+import alluxio.grpc.ScheduleAsyncPersistencePOptions;
+import alluxio.grpc.SetAclPOptions;
+import alluxio.grpc.SetAttributePOptions;
+import alluxio.grpc.TtlAction;
+import alluxio.grpc.UnmountPOptions;
+import alluxio.grpc.WritePType;
+
+/**
+ * This class contains static methods which can be passed Alluxio configuration objects that
+ * will populate the gRPC options objects with the proper values based on the given configuration.
+ */
+public class GrpcDefaultOptions {
+
+  /**
+   * @param conf Alluxio configuration
+   * @return options based on the configuration
+   */
+  public static CreateDirectoryPOptions getCreateDirectoryPOptions(AlluxioConfiguration conf) {
+    return CreateDirectoryPOptions.newBuilder()
+        .setCommonOptions(getFileSystemMasterCommonPOptions(conf))
+        .setWriteType(conf.getEnum(PropertyKey.USER_FILE_WRITE_TYPE_DEFAULT, WritePType.class))
+        // No default configuration values, will pull from gRPC defaultInstance
```

Comment:
I think it would be better to control non-configuration defaults from within the proto file rather than from these functions. Users don't necessarily have to go through this class to create these options and enforcing defaults from the proto definitions are a sure way to make sure that we always get the same defaults. Thoughts?

Reply:
We should make `getDefaultInstance` unavailable to end users. I think its better to control the defaults in one place rather than have some default in code and others in the proto definition. Also the proto def can be less flexible since it cannot be updated without recompilation.

Reply:
The issue I believe is that `getDefaultInstance()` isn't a function that we control. It's generated by protobuf. Obviously we could go in and manually change it to private but that means any time the file is regenerated we'd need to make that change. As far as I'm aware after glancing over the docs we can't control the availability of the `getDefaultInstance` function.

Reply:
We don't want to modify the autogenerated code. One soft method is to make `GrpcDefaultOptions` the public facing API. Users should only use the utils class when they need to get a default option, and if they decide to use the proto object directly they are on their own.

Reply:
Sorry I don't quite understand your suggestion. Are you saying that the client methods should take a different type instead (some kind of wrapper around the options)? or just that users should always use the `GrpcDefaultOptions` class?

Reply:
Users should always use the `GrpcDefaultOptions` class.